examples/hidden_shift_algorithm.py

```diff
@@ -0,0 +1,140 @@
+import random
+import cirq
+"""Demonstrates a Hidden Shift algorithm.
+
+Let f and g be two function oracles {0,1}^N -> {0,1}^N  which are the same
+up to a hidden bit string s:
+
+g(x) = f(x ⨁ s)
+
+The Hidden Shift Algorithm determines s by quering the two oracles. The
+implementation in this example considers the following oracle:
+
+f(x) = Σ_i x_(2i-1) x_(2i)
+
```

Comment:
Add more detail. Give a run through of this algorithm like our bb84 example does: https://github.com/quantumlib/Cirq/blob/master/examples/bb84.py.

Talk about how first the `Fourier` transform is computed, then the `f` oracle is called; the transform is called again, then g, then finally the inverse transform.

Reply:
Added a runthrough of the Algorithm. Added references.